packages/discovery-provider/src/tasks/index_payment_router.py

```diff
@@ -0,0 +1,858 @@
+                        # the the validator thinks the beginning of time is, and that's just too much work so I'm just not adding the blocktime filter in local dev
+                        query.filter(TrackPriceHistory.block_timestamp < timestamp)
+                    result = (
+                        query.filter(
+                            TrackPriceHistory.track_id == id,
+                        )
+                        .order_by(desc(TrackPriceHistory.block_timestamp))
+                        .first()
+                    )
+                    if result is not None:
+                        price = result.total_price_cents
+                        splits = result.splits
+                else:
+                    logger.error(
+                        f"index_payment_router.py | Unknown content type {type}"
+                    )
+                if (
+                    price is not None
+                    and splits is not None
+                    and isinstance(splits, dict)
+                    and content_owner_id is not None
+                ):
+                    return {
+                        "type": type,
+                        "id": id,
+                        "price": price * USDC_PER_USD_CENT,
+                        "splits": splits,
+                        "purchaser_user_id": purchaser_user_id,
+                        "content_owner_id": content_owner_id,
+                    }
+                else:
+                    logger.error(
+                        f"index_payment_router.py | Couldn't find relevant price for {content_metadata}"
+                    )
+            else:
+                logger.debug(
+                    f"index_payment_router.py | Ignoring memo, no content metadata found: {memo}"
+                )
+        except (ValueError, KeyError) as e:
+            logger.debug(
+                f"index_payment_router.py | Ignoring memo, failed to parse content metadata: {memo}, Error: {e}"
+            )
+    logger.error("index_payment_router.py | Failed to find any content metadata")
+    return None
+
+
+def validate_purchase(
+    purchase_metadata: PurchaseMetadataDict, balance_changes: dict[str, BalanceChange]
+):
+    """Validates the user has correctly constructed the transaction in order to create the purchase, including validating they paid the full price at the time of the purchase, and that payments were appropriately split"""
+    is_valid = True
+    # Check that the recipients all got the correct split
+    for account, split in purchase_metadata["splits"].items():
+        if account not in balance_changes or balance_changes[account]["change"] < split:
+            logger.error(
+                f"index_payment_router.py | Incorrect split given to account={account} amount={balance_changes[account]['change']} expected={split}"
+            )
+            is_valid = False
```

Comment:
nit, it could just be an early return of False here

Reply:
Done